### PR TITLE
[Refactor] Separate `_prepare_inputs` to `_prepare_inputs` and `_preprocess`

### PR DESCRIPTION
**Pull request by gcanlin**
### What this PR does / why we need it?

Part of RFC #5449.

Align with upstream vLLM. This PR will help downstream vLLM-Omni reduce the cost for maintaining the `_prepare_inputs`. Besides, it helps vLLM-Ascend code more readable. In the future, we can follow closer to vLLM.

- Moved the multimodal, prompt-embed, positions, PP handling and `update_cos_sin` into _preprocess, and trimmed `_prepare_input`s to return only metadata plus logits and spec-decode inputs.
- Updated `execute_model` to call `_prepare_inputs` then `_preprocess`, preserving the original ordering while separating concerns.
- Reuse `_prepare_mm_inputs` in vLLM and add `model_kwargs`.

NOTE: This PR includes #5971 changes. We need to wait it merged(if it would be approved). Then rebase this PR. 

### How was this patch tested?

- vLLM version: v0.13.0
- vLLM main: https://github.com/vllm-project/vllm/commit/2c24bc6996cb165fce92f780b388a5e39b3f4060

### CODE REVIEW

**Comment by gcanlin** [comment]
@wangxiyuan Could you please help add the ready tag for e2e-full test? I want to run it in the free weekend to avoid resource queuing. Thanks!

**Comment by gcanlin** [comment]
@zhenwenqi2024 Hi! Could you please help review this PR? Thx!

File: vllm_ascend/worker/model_runner_v1.py [hunk 1]
```diff
@@ -1153,6 +1034,142 @@ def _generate_process_reqs_hidden_states(self, maybe_padded_num_tokens,
...
+                    scheduler_output,
+                    encoder_cache=self.encoder_cache,
+            ):
+                # Run the multimodal encoder if any.
+                self._execute_mm_encoder(scheduler_output)
+
+                # NOTE(woosuk): To unify token ids and soft tokens (vision
+                # embeddings), we always use embeddings (rather than token ids)
+                # as input to the multimodal model, even when the input is text.
+                input_ids = self.input_ids.gpu[:num_scheduled_tokens]
```
Comment (gcanlin) [inline vllm_ascend/worker/model_runner_v1.py:1063]: Qwen-Omni in vLLM-Omni set `requires_raw_input_tokens` to keep input_ids value by `input_ids = self.input_ids.gpu[:num_tokens]`.

File: vllm_ascend/worker/model_runner_v1.py [hunk 2]
```diff
@@ -1153,6 +1034,142 @@ def _generate_process_reqs_hidden_states(self, maybe_padded_num_tokens,
...
+            )
+
+            # TODO(woosuk): Avoid the copy. Optimize.
+            self.inputs_embeds.gpu[:num_scheduled_tokens].copy_(inputs_embeds)
+
+            input_ids, inputs_embeds = self._prepare_mm_inputs(num_input_tokens)
+            if vllm_version_is('0.13.0'):
+                model_kwargs = {
+                    **self._init_model_kwargs(num_input_tokens),
+                    **self._extract_mm_kwargs(scheduler_output),
```
Comment (gcanlin) [inline vllm_ascend/worker/model_runner_v1.py:1080]: Qwen-Omni needs mm_kwargs
